### ISSUE TITLE
[Redstone Arsenal] Flux-Infused Armor provides its armor even when uncharged

### ISSUE THREAD

**Issue by Tora-B** [CoFH/Feedback#1120]
Flux-Infused Armor all provides its armor bonus when it has no charge. Charging it doesn't actually do anything, other than change the armor display on the HUD. Also, it's not possible to block unblockable damage anymore because of https://github.com/MinecraftForge/MinecraftForge/issues/3933, so there's a small block of dead code [here](https://github.com/CoFH/RedstoneArsenal/blob/5559754ab340cd6416d80e88f4123bd4f17b87f5/src/main/java/cofh/redstonearsenal/item/armor/ItemArmorFlux.java#L162-L165).

I'm not really sure what the proper way to do special armor is, and I've spent several hours trying to puzzle the whole mess out. ArmorMaterials don't directly affect armor or the HUD, but are typically used to create the Attributes, which affects the HUD, is displayed on tooltips, and used in damage calculations. ISpecialArmor#getProperties() is purely for calculation for a hit, and doesn't affect any display, which is why ISpecialArmor#GetArmorDisplay() even exists. Unfortunately, it only affects the HUD, and not tooltips, and it's additive to Attributes, rather than overriding. So if you want tooltips to display properly, you need to used Attributes, but then you lose control over when the armor is applied. If you only apply the attributes when the item should provide them, then the user loses information about the potential of the item. This may still be the best solution, and the item's potential armor (if charged) could be added to the tooltip manually, if desired.

My impression is that Attributes should be used for base armor values that apply to all hits, while ISpecialArmor#getProperties() should only be used for special circumstances, such as responding to the particular type or source of damage. Its original purpose appears to have been to apply a percentage-based damage calculation, separate from the vanilla damage calculation, and it still does that. However, that is in addition to the damage reduction from Attributes, and putting the full values in each results in the armor being twice as strong as intended.

So basically the options are to go as vanilla as possible, primarily using Attributes, and using getProperties() sparingly, or go full custom, stripping the default attributes, using getProperties() for all calculations, and using GetArmorDisplay() and custom tooltip display to make up the difference.

**Comment by KingLemming** [comment]
It used to be that ISpecialArmor fully superceded anything on the actual armor, but I know Forge has been mucking about in that arena lately.

Will look into it.

**Comment by KingLemming** [comment]
I have a mitigation for this but ehh...Forge has kind of borked the whole thing so it really isn't working quite like I want it to.

**Comment by Tora-B** [comment]
Now that I've had a little sleep (just a little!) and some time away from the code to think about it, I realize my initial report wasn't entirely accurate. The armor does provide full diamond-tier armor when uncharged, but charging it isn't pointless -- it provides an additional 90% absorption when charged. It's just hard to tell the difference at the relatively low damage values (3) I was testing at. This would explain the discrepancy between our results.
